File: Chroma-Core/src/main/java/buttondevteam/lib/chat/commands/CommandArgument.java
package buttondevteam.lib.chat.commands;

import lombok.RequiredArgsConstructor;
import org.javatuples.Pair;

/**
 * A command argument's information to be used to construct the command.
 */
@RequiredArgsConstructor
public class CommandArgument {
	public final String name;
	public final Class<?> type;
	public final boolean greedy;
	public final Pair<Double, Double> limits;
	public final boolean optional;
	public final String description;
}

File: Chroma-Core/src/main/java/buttondevteam/lib/chat/commands/CommandArgument.kt
package buttondevteam.lib.chat.commands

/**
 * A command argument's information to be used to construct the command.
 */
class CommandArgument(
    val name: String, // TODO: Remove <> from name and add it where appropriate
    val type: Class<*>,
    val greedy: Boolean,
    val limits: Pair<Double, Double>,
    val optional: Boolean,
    val description: String,
    val annotations: Array<Annotation>
)
